tools/AnalyzeDeps/AnalyzeDeps.ps1

```diff
@@ -0,0 +1,173 @@
+    $Locked[$DepName][$DepVersion].Add($Condition.Trim()) | Out-Null
+  }
+}
+
+# Analyze package dependencies
+$Pkgs = @{ }
+$Deps = @{ }
+foreach ($PkgFile in Resolve-Path $PackagesPath) {
+  $Nuspec = Get-Nuspec $PkgFile
+  $LibraryName = $Nuspec.package.metadata.id
+  $LibraryVer = $Nuspec.package.metadata.version
+
+  $Pkgs[$LibraryName] = @($LibraryVer, $PkgFile)
+
+  foreach ($Group in $Nuspec.package.metadata.dependencies.group) {
+    foreach ($Dep in $Group.dependency) {
+      Save-Dep $Deps $Group.targetFramework $Dep.id $Dep.version $LibraryName
+    }
+  }
+
+  foreach ($Dep in $Nuspec.package.metadata.dependencies.dependency) {
+    if ($Deps.Count) {
+      foreach ($TargetFramework in $Deps.Keys) {
+        Save-Dep $Deps $TargetFramework $Dep.id $Dep.version $LibraryName
+      }
+      Save-Dep $Deps "(others)" $Dep.id $Dep.version $LibraryName
+    }
+    else {
+      Save-Dep $Deps "(any)" $Dep.id $Dep.version $LibraryName
+    }
+  }
+}
+
+Write-Host "Analyzing $($Pkgs.Count) packages..."
+
+# Analyze lockfile
+$Locked = @{ }
+if ($LockfilePath) {
+  [xml]$PackageProps = Get-Content $LockfilePath
+  foreach ($ItemGroup in $PackageProps.Project.ItemGroup) {
+    if ($ItemGroup.Condition) {
+      $Condition = $ItemGroup.Condition
+    } else {
+      $Condition = ""
+    }
+    foreach ($Entry in $ItemGroup.PackageReference) {
+      if ($Entry.Update) {
+        Save-Locked $Locked $Entry.Update $Entry.Version $Condition
+      }
+    }
+  }
+  Write-Host "Discovered $($Locked.Count) versions pinned in the lockfile."
+} else {
+  Write-Warning "No lockfile was provided, or the lockfile was empty. Declared dependency versions were not able to be validated against the lockfile."
+}
+
+# Precompute some derived data for the template
+$External = $Deps.Keys | Where-Object { -not ($Pkgs.ContainsKey($_)) }
+$Inconsistent = @{ }
+$MismatchedVersions = @{ }
+$Unlocked = New-Object System.Collections.ArrayList
+foreach ($DepName in $Deps.Keys) {
+  $InconsistentFrameworks = $Deps[$DepName].Keys | Where-Object { $Deps[$DepName][$_].Count -gt 1 }
+  if ($InconsistentFrameworks) {
+    $Inconsistent[$DepName] = $InconsistentFrameworks
+  }
+
+  if ($Locked.ContainsKey($DepName)) {
+    foreach ($TargetFramework in $Deps[$DepName].Keys) {
+      foreach ($DepVer in $Deps[$DepName][$TargetFramework].Keys) {
+        if (-Not $Locked[$DepName].ContainsKey($DepVer)) {
+          if (-Not $MismatchedVersions[$DepName]) {
+            $MismatchedVersions[$DepName] = @{ }
+          }
+          $MismatchedVersions[$DepName][$DepVer] = $Deps[$DepName][$TargetFramework][$DepVer]
+        }
+      }
+    }
+  } else {
+    $Unlocked.Add($DepName) | Out-Null
+  }
+}
+
+$ExitCode = 0
+if ($Inconsistent) {
+  Write-Warning "$($Inconsistent.Count) inconsistent dependency versions were discovered."
+  # Don't fail the build when inconsistent dependencies are present
+  # TODO: Remove this ASAP
```

Comment:
Is there an issue tracking this TODO?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/6771